mlir/include/mlir/Dialect/Transform/IR/TransformOps.td

```diff
@@ -691,6 +691,65 @@ def GetTypeOp : TransformDialectOp<"get_type",
                        "functional-type(operands, results)";
 }
 
+def HoistLoopInvariantSubsetsOp
```

Comment:
I'm not sure I fully endorse this op living in the "core" transform dialect. I understand that it is interface-based, so it doesn't belong with any specific dialect, but that doesn't sound like a great way to group things. IMO, the "core" part should mostly contain generic constructs like `foreach` and `apply_patterns` that don't depend even on interfaces. For the rest, we can have TD extensions living within the TD itself, we already do for PDL patterns. Though I can't come up with a good name. WDYT?

Reply:
Makes sense. We possible extensions that come to mind:
- an extension for "transformations of loop-like ops": `apply_licm`, `hoist_loop_invariant_subsets`
- an extension for all "MLIR core transformations": `apply_licm`, `apply_patterns`, `apply_conversion_pattern`, `hoist_loop_invariant_subsets`, `apply_dce`, maybe I missed a few more

I would prefer the second one, what do you think? We could call it `Dialect/Transform/CoreTransformsExtension/CoreTransformsExtension.cpp`. (Or if you know a better name...)

Reply:
Somehow, I don't put `apply_patterns` and `apply_dce` into the same bucket as `apply_licm` and `hoist_loop_invariant_subsets`. Maybe because the former are supposed to work on ops from _all_ dialects, and the latter only apply to a subset of dialects. So I'd prefer a loop-specific extension. The naming issue is that SCF-related transforms already use `loop` as a prefix (maybe we should change that).